tests/core/pyspec/eth2spec/test/context.py

```diff
@@ -48,28 +50,46 @@ class SpecForks(TypedDict, total=False):
     PHASE1: SpecPhase1
 
 
+def _prepare_state(balances_fn: Callable[[Any], Sequence[int]], threshold_fn: Callable[[Any], int],
+                   spec: Spec, phases: SpecForks):
+
+    p0 = phases[PHASE0]
+    balances = balances_fn(p0)
+    activation_threshold = threshold_fn(p0)
+
+    state = create_genesis_state(spec=p0, validator_balances=balances,
+                                 activation_threshold=activation_threshold)
+    if spec.fork == PHASE1:
+        # TODO: instead of upgrading a test phase0 genesis state we can also write a phase1 state helper.
+        # Decide based on performance/consistency results later.
+        state = phases[PHASE1].upgrade_to_phase1(state)
+        # Shard state slot must lag behind BeaconState slot by at least 1
+        # Will handle this more elegantly with fork mechanics
+        spec.process_slots(state, state.slot + 1)
+
+    return state
+
+
+_custom_state_cache_dict = LRU(size=10)
```

Comment:
any particular reason why 10?

Reply:
- 1 is too small, since config or phase can change back and forth.
- 4 would just about fit.
- 10 seemed safe enough, and behavior seems to match (getting a lot of cache hits for same config / phase afterwards)